sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/Utilities/UtilityAnalyzerBase.cs

```diff
@@ -44,6 +44,8 @@ public abstract class UtilityAnalyzerBase : SonarDiagnosticAnalyzer
 
         protected string WorkDirectoryBasePath { get; set; }
 
+        protected virtual bool AnalyzeAutogeneratedCode { get; set; }

```

Comment:
all utility analyzers except the `FileMetadataAnalyzer` were ignoring generated code. This change makes them read the setting value and generate the metrics/cpd/symbol/token metadata for generated files as well